test/Tizen.NUI.WindowSystem.InputGesture/Tizen.NUI.WindowSystem.InputGesture.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using Tizen;
+using Tizen.NUI;
+using Tizen.NUI.BaseComponents;
+using Tizen.NUI.WindowSystem;
+using System.Collections.Generic;
+
+namespace Tizen.NUI.WindowSystem
+{
+    class Program : NUIApplication
+    {
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+            Initialize();
+        }
+
+        void Initialize()
+        {
+            Window win = Window.Instance;
+            inputGesture = new InputGesture();
+
+            win.WindowSize = new Size2D(500, 500);
+            win.KeyEvent += OnKeyEvent;
+            win.BackgroundColor = Color.White;
+
+            View windowView = new View();
+            windowView.Size2D = new Size2D(500, 500);
+            windowView.BackgroundColor = Color.White;
+            windowView.TouchEvent += OnTouchEvent;
+            win.Add(windowView);
+
+            centerLabel = new TextLabel("InputGesture Sample, Click to generate Return Key.");
+            centerLabel.HorizontalAlignment = HorizontalAlignment.Center;
+            centerLabel.VerticalAlignment = VerticalAlignment.Center;
+            centerLabel.TextColor = Color.Black;
+            centerLabel.PointSize = 12.0f;
+            centerLabel.HeightResizePolicy = ResizePolicyType.FillToParent;
+            centerLabel.WidthResizePolicy = ResizePolicyType.FillToParent;
+            windowView.Add(centerLabel);
+
+            repeatCounter = 0;
+        }
+
+        private void OnKeyEvent(object sender, Window.KeyEventArgs e)
+        {
+            if (e.Key.State == Key.StateType.Down && (e.Key.KeyPressedName == "XF86Back" || e.Key.KeyPressedName == "Escape"))
```

Comment:
Using IntPtr in user code is generally discouraged. If feasible, I suggest either creating a new class that wraps the native handle using either HandleRef or SafeHandle, or modifying the gesture class to perform such handling itself.
but I'm just stating my own opinion here, so you can consider it.